display requests cards after reloading the /requests page

pankajjs opened the pull request:
<!-- Date Here in dd mmm yyyy-->
Date: 7 April, 2025
<!--Developer Name Here-->
Developer Name: @pankajjs 

---

## Issue Ticket Number
<!--Issue ticket this PR closes-->
- Closes #972 
## Description
This PR fixes an issue where error UI toast were being shown in /requests page after refreshing the page. Now, it correctly shows the requests card and hides the error UI toast after refreshing the page.
<!--Description of the changes made in this PR-->

### Documentation Updated?

- [ ] Yes
- [x] No

<!--Additional notes about documentation update if applicable-->

### Under Feature Flag

- [ ] Yes
- [x] No

<!--Indicate if changes are under a feature flag-->

### Database Changes

- [ ] Yes
- [x] No

<!--Notes on any database changes-->

### Breaking Changes

- [ ] Yes
- [x] No

<!--Notes on breaking changes or related issue tickets if applicable-->

### Development Tested?

- [x] Yes
- [ ] No

<!--Confirmation of local testing during development-->

## Screenshots
<details>
<summary>Working Video</summary>


https://github.com/user-attachments/assets/a776fc4f-e9a2-4d69-a412-f1093290081c


<!-- Attach your screenshots here👇-->

</details>

<!--Attach or link to relevant screenshots or visual aids-->

## Test Coverage
<details>
<summary>Screenshots</summary>
<img width="942" alt="Screenshot 2025-04-09 at 8 46 58 PM" src="https://github.com/user-attachments/assets/e9168837-2c8a-44dd-8fde-6ed757a5cee7" />

<img width="887" alt="Screenshot 2025-04-10 at 2 29 13 PM" src="https://github.com/user-attachments/assets/3cf3254a-a1ad-471e-87c2-9448540ed14b" />

<img width="828" alt="Screenshot 2025-04-08 at 5 25 13 PM" src="https://github.com/user-attachments/assets/14bc583e-4d3c-41d2-872b-e73199e17d4a" />

<!-- Attach your screenshots here👇-->

</details>

<!--Attach Details on test coverage and outcomes-->

## Additional Notes

<!--Any additional notes, considerations or explanations for reviewers-->

pankajjs commented on requests/script.js:
```diff
@@ -19,7 +19,7 @@ const filterOptionsContainer = document.getElementById(
 const applyFilterButton = document.getElementById('applyFilterButton');
 const userNameFilterInput = document.getElementById('assignee-search-input');
-let currentReqType = params.get('type') ?? OOO_REQUEST_TYPE;
+let currentReqType = params.get('type')?.toUpperCase() ?? OOO_REQUEST_TYPE;
```
API uses type in capital letters. Therefore converting request type to upper case just after fetching from url.

MayankBansal12 replied:
Any specific reason for not using `const` here?

MayankBansal12 commented:
@pankajjs Please include test coverage as well

pankajjs commented on requests/script.js:
```diff
@@ -81,7 +81,7 @@ oooTabLink.addEventListener('click', async function (event) {
   deselectRadioButtons();
   userNameFilterInput.value = '';
-  updateTabLink(currentReqType.toUpperCase());
+  updateTabLink(currentReqType);
```
Un-necessary as currentType is in capital letter.
note: removed for all updateTabLink call

pankajjs commented:
@coderabbitai review

pankajjs commented on __tests__/requests/requests.test.js:
```diff
@@ -155,7 +167,11 @@ describe('Tests the request cards', () => {
   });
 
   it('should update the card when the accept or reject button is clicked for OOO requests', async () => {
+    await page.goto(`${LOCAL_TEST_PAGE_URL}/requests`);
+    await page.waitForNetworkIdle();
+
     await page.click('#ooo_tab_link');
+    expect(page.url()).toContain('type=ooo');
```
It was failing because of different requests page url. Therefore, made the browser go to default requests page and then clicked on ooo_tab_link. To make sure it is working, expected the correct page url on line 174.

AnujChhikara commented on __tests__/requests/requests.test.js:
```diff
@@ -129,6 +129,18 @@ describe('Tests the request cards', () => {
     await browser.close();
   });
 
+  it('should not show the error toast ui after reloading a page', async () => {
+    await page.click('#extension_tab_link');
+    expect(page.url()).toContain('type=extension');
+    await page.reload();
+    expect(page.url()).toContain('type=extension');
```
Hello @pankajjs this test case is great addition but can we add another test where we test that when we reload the page we are loading the correct content?

prakashchoudhary07 commented:
@pankajjs If this is a test fix, can we please update the PR title
From the title, I understand we are fixing a bug and were trying to find the code for it. Then saw the description to know it was a test PR. So let's have a proper name(You can also follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) )

pankajjs commented:
> @pankajjs If this is a test fix, can we please update the PR title From the title, I understand we are fixing a bug and were trying to find the code for it. Then saw the description to know it was a test PR. So let's have a proper name(You can also follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) )

Hello @prakashchoudhary07 This is a bug fix. However, I have fixed existing test case in this PR because it was failing to my changes(refering to this comment https://github.com/Real-Dev-Squad/website-dashboard/pull/973#discussion_r2033059744). The code changes are in index.html and script.js under request folder.
Attached Screenshots -
<img width="1116" alt="Screenshot 2025-04-12 at 12 48 55 PM" src="https://github.com/user-attachments/assets/394e23bf-d4f2-43ba-aae1-c9b2c3c90739" />
<img width="1047" alt="Screenshot 2025-04-12 at 12 49 14 PM" src="https://github.com/user-attachments/assets/72a42e4d-c982-4038-9db3-f27848f175ea" />

let me know If I was able to address your concerns.

prakashchoudhary07 commented:
> > @pankajjs If this is a test fix, can we please update the PR title From the title, I understand we are fixing a bug and were trying to find the code for it. Then saw the description to know it was a test PR. So let's have a proper name(You can also follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) )
> 
> Hello @prakashchoudhary07 This is a bug fix. However, I have fixed existing test case in this PR because it was failing to my changes(refering to this comment [#973 (comment)](https://github.com/Real-Dev-Squad/website-dashboard/pull/973#discussion_r2033059744)). The code changes are in index.html and script.js under request folder. Attached Screenshots - <img alt="Screenshot 2025-04-12 at 12 48 55 PM" width="1116" src="https://private-user-images.githubusercontent.com/149887791/433018924-394e23bf-d4f2-43ba-aae1-c9b2c3c90739.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ0OTk2MDMsIm5iZiI6MTc0NDQ5OTMwMywicGF0aCI6Ii8xNDk4ODc3OTEvNDMzMDE4OTI0LTM5NGUyM2JmLWQ0ZjItNDNiYS1hYWUxLWM5YjJjM2M5MDczOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMlQyMzA4MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kODlhMTI3NDczMGM0YTg2OTU1YjBkMDM0NzNkYTRhYWIwYzcyMzVhZDhiM2MwMmM1YzRkOTk0MTIyZDZhN2JmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cWLfgN45cOmJRXiPLKvKeROvoozw_GmBguCrhWnU-Ec"> <img alt="Screenshot 2025-04-12 at 12 49 14 PM" width="1047" src="https://private-user-images.githubusercontent.com/149887791/433018925-72a42e4d-c982-4038-9db3-f27848f175ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ0OTk2MDMsIm5iZiI6MTc0NDQ5OTMwMywicGF0aCI6Ii8xNDk4ODc3OTEvNDMzMDE4OTI1LTcyYTQyZTRkLWM5ODItNDAzOC05ZGIzLWYyNzg0OGYxNzVlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMlQyMzA4MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03Y2FlNDE4NzVjMjJlY2U2MGVjMTU3NDQwMzJjYmJmM2JkMmUyMGEyZWM1MzI0ZTgzM2EyMzQwYmNkMjI3MmU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-5CRNxxC-DD8aaVkBdITy6rcBkpK8C_37w-MPobSE4g">
> 
> let me know If I was able to address your concerns.

So there was some issues with the code, and adding "data-testid=toast" fixed it right that is the conclusion correct?

prakashchoudhary07 commented on requests/index.html:
```diff
@@ -57,7 +57,7 @@
           >
         </nav>
-        <div id="toast" class="hidden">
+        <div id="toast" class="hidden" data-testid="toast">
```
Where is `data-testid="toast"` being used?

pankajjs replied:
<img width="986" alt="Screenshot 2025-04-13 at 12 45 49 PM" src="https://github.com/user-attachments/assets/b2f3ebb0-1f07-4335-99ab-5953d6e8332e" />

@prakashchoudhary07 the data-testid=toast is being used in this test at line 138.

pankajjs commented:
> > > @pankajjs If this is a test fix, can we please update the PR title From the title, I understand we are fixing a bug and were trying to find the code for it. Then saw the description to know it was a test PR. So let's have a proper name(You can also follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) )
> > 
> > 
> > Hello @prakashchoudhary07 This is a bug fix. However, I have fixed existing test case in this PR because it was failing to my changes(refering to this comment [#973 (comment)](https://github.com/Real-Dev-Squad/website-dashboard/pull/973#discussion_r2033059744)). The code changes are in index.html and script.js under request folder. Attached Screenshots - <img alt="Screenshot 2025-04-12 at 12 48 55 PM" width="1116" src="https://private-user-images.githubusercontent.com/149887791/433018924-394e23bf-d4f2-43ba-aae1-c9b2c3c90739.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ0OTk2MDMsIm5iZiI6MTc0NDQ5OTMwMywicGF0aCI6Ii8xNDk4ODc3OTEvNDMzMDE4OTI0LTM5NGUyM2JmLWQ0ZjItNDNiYS1hYWUxLWM5YjJjM2M5MDczOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMlQyMzA4MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kODlhMTI3NDczMGM0YTg2OTU1YjBkMDM0NzNkYTRhYWIwYzcyMzVhZDhiM2MwMmM1YzRkOTk0MTIyZDZhN2JmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cWLfgN45cOmJRXiPLKvKeROvoozw_GmBguCrhWnU-Ec"> <img alt="Screenshot 2025-04-12 at 12 49 14 PM" width="1047" src="https://private-user-images.githubusercontent.com/149887791/433018925-72a42e4d-c982-4038-9db3-f27848f175ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ0OTk2MDMsIm5iZiI6MTc0NDQ5OTMwMywicGF0aCI6Ii8xNDk4ODc3OTEvNDMzMDE4OTI1LTcyYTQyZTRkLWM5ODItNDAzOC05ZGIzLWYyNzg0OGYxNzVlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMlQyMzA4MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03Y2FlNDE4NzVjMjJlY2U2MGVjMTU3NDQwMzJjYmJmM2JkMmUyMGEyZWM1MzI0ZTgzM2EyMzQwYmNkMjI3MmU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-5CRNxxC-DD8aaVkBdITy6rcBkpK8C_37w-MPobSE4g">
> > let me know If I was able to address your concerns.
> 
> So, there were some issues with the code, and adding "data-testid=toast" fixed them, right? That is the conclusion, correct?

Added data-testid=toast to test the behaviour also there are changes in script file.
* Attached screenshot of data-testId being used at line 138.
<img width="986" alt="Screenshot 2025-04-13 at 12 45 49 PM" src="https://github.com/user-attachments/assets/24181560-a389-4974-baf1-947630c046c0" />

* Attached screenshots of changes in script file which fixes the bug. 
- Change at line 22
<img width="864" alt="Screenshot 2025-04-13 at 12 45 59 PM" src="https://github.com/user-attachments/assets/2b1fecc6-3d57-4d10-aaa2-3365d3cd6d2a" />

- Change at line 84
<img width="864" alt="Screenshot 2025-04-13 at 12 46 04 PM" src="https://github.com/user-attachments/assets/cb31cd49-22b9-4827-85f2-3e9a3af986b2" />

- Change at line 97 and 110
<img width="848" alt="Screenshot 2025-04-13 at 12 46 08 PM" src="https://github.com/user-attachments/assets/9ce1c584-8fc1-454c-b78f-6bf9fab76aaa" />
- Change at line 781
<img width="1115" alt="Screenshot 2025-04-13 at 12 46 12 PM" src="https://github.com/user-attachments/assets/0ea8361a-a979-4c13-b1f3-22e5b50f7656" />

pankajjs commented on __tests__/requests/requests.test.js:
```diff
@@ -208,6 +224,37 @@ describe('Tests the request cards', () => {
   });
 
+  it('should show requests cards after reloading the page', async () => {
+    await page.goto(`${LOCAL_TEST_PAGE_URL}/requests`);
+    await page.waitForNetworkIdle();
+
+    await page.click('#extension_tab_link');
+    expect(page.url()).toContain('type=extension');
+
```
https://github.com/Real-Dev-Squad/website-dashboard/pull/973/files#diff-e334709afd081cd32a58d0804717da01de3ff85928de5d3039aa361ea7bbd021R228-R233

devprakash0 commented on requests/script.js:
```diff
@@ -698,7 +716,10 @@ userNameFilterInput.addEventListener(
 
-            const requestData = { requestedBy: user.username };
+            const requestData = {
+              requestedBy: user.username,
```
can we add null check here

devprakash0 commented on requests/script.js:
```diff
@@ -470,7 +483,7 @@ async function renderRequestCards(queries = {}) {
       userDetails = await getInDiscordUserList();
     }
-    requestResponse = await getRequests(currentReqType, queries);
+    requestResponse = await getRequests(queries.requestType, queries);
```
same here
